### PR TITLE
protect API email & SMS routes

### PR DESCRIPTION
**Pull request by haroon-ali-dev**
#48 Only requests sent from the application will now be accepted.

### CODE REVIEW

**Review by Miladepour — APPROVED** [review]
Thanks, Haroon, I finished my review and tested it.

File: client/src/pages/result/Result.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ export default function Result() {
 			method: "POST",
 			headers: {
 				"Content-Type": "application/json",
+				"Request-Source": 1
```
Comment (EneVladMihai) [inline client/src/pages/result/Result.js:51]: Could you please explain / link some docs about the approach?

I think the idea was to use the request's origin and have a list of allowed hosts in the backend which we validate.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin

The Origin header can not be modified programmatically. But you could put "Request-Source" header in postman and make the request successfully which is not what we want.

**Review by EneVladMihai — CHANGES_REQUESTED** [review]

File: client/src/pages/result/Result.js (hunk 1 above)
Comment (haroon-ali-dev) [reply client/src/pages/result/Result.js]: That was a solution that just came to mind. Your solution makes a lot more sense.
I've implemented it.

**Review by EneVladMihai — CHANGES_REQUESTED** [review]
Avoid hardcoding and remove debug logs

File: server/middlewares/requestSource.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+function requestSource(req, res, next) {
+  if (req.get("origin") === "http://localhost:3000" || req.get("origin" === "https://starter-kit-j5ar.onrender.com")) {
+    console.log("Hello World");
```
[the remark this answers is not in the document]
Comment (haroon-ali-dev) [reply server/middlewares/requestSource.js]: Done.

File: server/middlewares/requestSource.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+function requestSource(req, res, next) {
+  if (req.get("origin") === "http://localhost:3000" || req.get("origin" === "https://starter-kit-j5ar.onrender.com")) {
```
[the remark this answers is not in the document]
Comment (haroon-ali-dev) [reply server/middlewares/requestSource.js]: Done.

**Review by EneVladMihai — APPROVED** [review]

**Pull request merged**